app/src/main/java/com/rednerracaza/homework/view/ForecastDetailsPage.kt

```diff
@@ -0,0 +1,64 @@
+package com.rednerracaza.homework.view
+
+import android.os.Bundle
+import android.view.View
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.navArgs
+import com.rednerracaza.homework.R
+import com.rednerracaza.homework.databinding.FragmentForecastDetailsBinding
+import com.rednerracaza.homework.util.Others.celsiusSymbol
+import com.rednerracaza.homework.util.Others.getCityStatus
+import com.rednerracaza.homework.util.Others.setData
+import com.rednerracaza.homework.viewModel.ForecastPageViewModel
+import dagger.hilt.android.AndroidEntryPoint
+
+/**
+ * [ForecastDetailsPage] - this fragment page shows the details of forecast selected recyclerview from api/server
+ *
+ * [viewModel] - this viewModel is shared to ForecastPage fragment.
+ *
+ * [_binding] - viewBinding for this layout
+ *
+ * [lifecycleScope] - this lifecycleScope is tied to this fragment, if this fragment is destroyed it will cancelled it self.
+ *
+ * [onDestroy] - will set _binding to null to free some memory
+ *
+ * [onStart] - will request to the server with Id
+ *
+ */
+@AndroidEntryPoint
+class ForecastDetailsPage : Fragment(R.layout.fragment_forecast_details) {
+    private val viewModel by activityViewModels<ForecastPageViewModel>()
+    private val args: ForecastDetailsPageArgs by navArgs()
```

Comment:
For uniformity declaration of variable types I suggest to do the same code as you did in the variable viewModel.
```suggestion
    private val args by navArgs<ForecastDetailsPageArgs >()
```

Reply:
Noted sir thank you.